server/plugin/webhook.go

```diff
@@ -147,13 +147,25 @@ func (wb *WebhookBroker) Close() {
 func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
 	config := p.getConfiguration()
 
-	signature := r.Header.Get("X-Hub-Signature")
-	body, err := ioutil.ReadAll(r.Body)
+	body, err := io.ReadAll(r.Body)
```

Comment:
I'm not sure what's happening here; why is this any better than the original? We're still reading the whole body regardless of size, right?

Reply:
Yes, this PR doesn't change how the body is read, But it changes how the body is parsed. We now check the HMAC before calling `github.ParseWebHook`.

Reply:
Got it; and I think that's fine and a good defense-in-depth improvement, knowing that there is a separate size limit in the server.